src/main/java/sirius/kernel/cache/CacheRemoverBuilder.java

```diff
@@ -0,0 +1,15 @@
+package sirius.kernel.cache;
+
+import java.util.function.BiFunction;
+import java.util.function.BiPredicate;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+public interface CacheRemoverBuilder<K, V, T> {
+    <R> CacheRemoverBuilder<K, V, R> map(BiFunction<String, T, R> mapper);
+
+    Cache<K, V> removeIf(BiPredicate<String, T> predicate);
+    <R> CacheRemoverBuilder<K, V, R> map(Function<T, R> mapper);
+
+    Cache<K, V> removeIf(Predicate<T> predicate);
```

Comment:
probably unused as the discriminator is missing (?)

Reply:
Same as above. Its very little implementation overhead, and it does allow some flexibility, like a `removeIf(Objects::nonNull)`. The discriminator might have been used earlier.

However, it might be a good idea to force the api user to ignore a parameter, he might be tempted to forget the descriminator totally